### PR TITLE
Batch Processing with Multiple tasks in a task-graph

### PR DESCRIPTION
**Pull request by jjfumero**
#### Description

This fixes and adds support for batch processing when having a `TaskGraph` with multiple tasks. 
Besides, it removes old/unreachable methods that are not needed anymore. 

However, this PR still has the issue of incorrect results when the task-graph does not perform a copy-in. 
I added two more unit-tests, one with copy in and the other without. It seems that without copy-in, the batch processing works (compiler and internals of the runtime), but the results are not correct. 

For task-graphs with copy-in/out (which was the original intended use), multiple tasks in a taskgraph with batch processing now works. 
We can go ahead this is partial fix, and add a new extension on top of this to process also for only outputs. 

#### Problem description

The issue was that, when having multiple tasks in a task-graph, the TornadoVM interpreter was not invalidating the code cache (to force recompilation). 

#### Backend/s tested

Mark the backends affected by this PR.

- [X] OpenCL
- [ ] PTX
- [ ] SPIRV

#### OS tested

Mark the OS where this PR is tested.

- [X] Linux
- [ ] OSx
- [ ] Windows

#### Did you check on FPGAs?

If it is applicable, check your changes on FPGAs.

- [ ] Yes
- [X] No

#### How to test the new patch?

```bash
$ make
$ tornado-test -V --fast --debug --threadInfo --printKernel --printBytecode uk.ac.manchester.tornado.unittests.batches.TestBatches#testBatchNotEven
$ tornado-test -V --fast --debug --threadInfo --printKernel --printBytecode uk.ac.manchester.tornado.unittests.batches.TestBatches#testBatchNotEven2
```

### CODE REVIEW

File: tornado-runtime/src/main/java/uk/ac/manchester/tornado/runtime/tasks/TornadoTaskGraph.java [hunk 1]
```diff
@@ -542,22 +542,6 @@ public void setDevice(String taskName, TornadoDevice device) {
 
     }
 
-    private void triggerRecompile() {
```
Comment (mikepapadim) [inline tornado-runtime/src/main/java/uk/ac/manchester/tornado/runtime/tasks/TornadoTaskGraph.java:545]: What happend with this ?

**Review by mikepapadim — APPROVED** [review]
LGMT

**Review by mairooni — APPROVED** [review]

File: tornado-runtime/src/main/java/uk/ac/manchester/tornado/runtime/tasks/TornadoTaskGraph.java (hunk 1 above)
Comment (jjfumero) [reply tornado-runtime/src/main/java/uk/ac/manchester/tornado/runtime/tasks/TornadoTaskGraph.java:545]: This method was not used.

Comment (mairooni) [reply tornado-runtime/src/main/java/uk/ac/manchester/tornado/runtime/tasks/TornadoTaskGraph.java:545]: This was never used

**Pull request merged**